R/checks.R

```diff
@@ -120,11 +120,18 @@ CheckForUniqueVariableNames <- function(formula)
 #' @param object A model object for which prediction is desired.
 #' @param newdata A \code{data.frame} including the variables used to fit the model.
 #' @importFrom flipU CopyAttributes
+#' @importFrom flipFormat Labels
 #' @export
 CheckPredictionVariables <- function(object, newdata)
 {
-    # Check that newdata contains all training variables
-    training <- object$model[object$subset, names(object$model) != object$outcome.name, drop = FALSE]
+    regression.model <- inherits(object, "Regression")
+    relevant.cols <- names(object$model)[names(object$model) != object$outcome.name]
+    # Check if a regression object is being processed and the outlier removal has been implemented.
+    if (outliers.removed <- (regression.model && !all(non.outliers <- object$non.outlier.data)))
```

Comment:
I do it too sometimes, but I'm not a big of variable assignments in `if` statements because I think it makes the code less readable.

Reply:
I'll move the assignment outside